@here/harp-examples/lib/PerformanceUtils.ts

```diff
@@ -0,0 +1,829 @@
+        dataSourceType: string[],
+        decoderCount?: number,
+        phasedLoading?: boolean,
+        powerPreference?: MapViewPowerPreference,
+        storageLevelOffsetModifier: number = 0,
+        theme: ThemeDef = DEFAULT_THEME
+    ): Promise<MapViewApp> {
+        const mapViewApp = initializeMapViewApp(
+            id,
+            decoderCount,
+            phasedLoading,
+            powerPreference,
+            theme
+        );
+
+        // Store time MapView has been initialized
+        const appInitTime = PerformanceTimer.now();
+
+        // Set to `true` to visualize the text placement collisions
+        debugContext.setValue("DEBUG_SCREEN_COLLISIONS", false);
+
+        return new Promise<MapViewApp>((resolve, reject) => {
+            const dataSourceInitialized = connectDataSources(
+                mapViewApp,
+                dataSourceType,
+                storageLevelOffsetModifier
+            );
+
+            dataSourceInitialized
+                .then(() => {
+                    PerformanceStatistics.instance.appResults.set("startTime", appStartTime);
+                    PerformanceStatistics.instance.appResults.set("initTime", appInitTime);
+                    resolve(mapViewApp);
+                })
+                .catch(err => {
+                    reject(new Error("Failed to initialize WARP datasource"));
+                });
+        });
+    }
+
+    function connectDataSources(
+        mapViewApp: MapViewApp,
+        dataSourceTypes: string[],
+        storageLevelOffsetModifier: number
+    ): Promise<DataSource[]> {
+        let dataSource: OmvDataSource | undefined;
+
+        const hereCopyrightInfo: CopyrightInfo = {
+            id: "here.com",
+            year: new Date().getFullYear(),
+            label: "HERE",
+            link: "https://legal.here.com/terms"
+        };
+        const copyrights: CopyrightInfo[] = [hereCopyrightInfo];
+
+        const promises: Array<Promise<DataSource>> = dataSourceTypes.map(dataSourceType => {
+            switch (dataSourceType) {
+                case "OMV":
+                    dataSource = new OmvDataSource({
+                        baseUrl: "https://xyz.api.here.com/tiles/herebase.02",
+                        apiFormat: APIFormat.XYZOMV,
+                        styleSetName: "tilezen",
+                        maxZoomLevel: 17,
+                        authenticationCode: accessToken,
+                        copyrightInfo: copyrights
+                    });
+                    break;
+                default:
+                    throw new Error("Unknown data source");
+            }
+
+            if (storageLevelOffsetModifier !== undefined && storageLevelOffsetModifier !== 0) {
+                dataSource.storageLevelOffset =
+                    dataSource.storageLevelOffset + storageLevelOffsetModifier;
+            }
+
+            return new Promise(resolve => {
+                const connected = () => {
```

Comment:
This code was breaking the decoder test, I believe it was caused by the addition of the background tile source. See the fix in https://gerrit.it.here.com/#/c/1181047/

Reply:
Fixed